_drafts/bazel-for-embedded.md

```diff
@@ -0,0 +1,383 @@
+
+In our case, the constraints for the execution platform will describe the host on which the toolchain we downloaded is built to run. While the constraints for the target platform will describe the ARM CPU.
+
+Here what the toolchain rule can look like in `toolchain_arm64_gcc/BUILD`:
+
+```python
+toolchain(
+    name = "arm64_gcc_linux_x86_64",
+    exec_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+    target_compatible_with = [
+        "@platforms//cpu:arm64",
+    ],
+    toolchain_type = "@rules_cc//cc:toolchain_type",
+    toolchain = "@arm64_gcc_linux_x86_64//:cc_toolchain",
+)
+```
+
+#### Register the toolchain
+
+Now we need to register the toolchain so that Bazel knows about its existence. This is done in the `WORKSPACE` file as follow:
+
+```python
+register_toolchains(
+    "//toolchain_arm64_gcc:arm64_gcc_linux_x86_64",
+)
+```
+
+#### Create a platform
+
+At this point, the toolchain is fully integrated into Bazel and ready to be used. We only need a way to select it, this is done by creating a platform that will set the constraints we defined previously in the toolchain rule.
+
+We need to create a platform that will be used as the target platform. Note the execution platform is the same as the host platform and is auto generated by Bazel and it matches your current host environment.
+
+This target platform only need to set constraints to define the ARM CPU, it can be done as follow in `toolchain_arm64_gcc/BUILD`:
+
+```python
+platform(
+    name = "toolchain_arm64_gcc",
+    constraint_values = [
+        "@platforms//cpu:arm64",
+    ],
+)
+```
+
+#### Use the toolchain
+
+We now have all the pieces to build a target with our new toolchain. To invoke this toolchain, we need to tell Bazel that we want to build for the previously create target platform. This done as follow:
+
+```bash
+bazel build … --platforms=//toolchain_arm64_gcc --incompatible_enable_cc_toolchain_resolution
+```
+
+This is it! You have integrated a semi-hermetic toolchain into Bazel!
+
+Note, `--incompatible_enable_cc_toolchain_resolution` is required to enable toolchain resolution with platforms for C++ toolchains. This flag should be flipped in future Bazel releases and therefore not be needed anymore.
+
+The code described in this section can be found here: https://github.com/blaizard/bazel-examples/tree/main/toolchain-arm64-gcc
+
+### Dockerized
+
+Often when dealing with cross compilation, you don’t get a self contained binary or package that you can just unzip and use directly, but you need to install the toolchain on your machine, which makes it very difficult to use or integrate in CI for example or might conflict with an existing installation you already have.
```

Comment:
Might be good to touch on how this can make it easier for developers to set up their environment as well.